app/src/marketplace/components/Modal/ChooseAccessPeriodDialog/index.jsx

```diff
@@ -7,255 +7,222 @@ import cx from 'classnames'
 
     const [priceInData, priceInUsd] = useMemo(() => {
-        const pricePerSecondInData = priceCurrency === contractCurrencies.DATA ?
-            pricePerSecond :
-            usdToData(pricePerSecond, dataPerUsd)
-
-        const pricePerSecondInUsd = priceCurrency === contractCurrencies.USD ?
-            pricePerSecond :
-            dataToUsd(pricePerSecond, dataPerUsd)
+        const inData = dataForTimeUnits(
+            pricePerSecond,
+            dataPerUsd,
+            priceCurrency,
+            time,
+            timeUnit,
+        )
 
         return [
-            getPrice(time, timeUnit, pricePerSecondInData, contractCurrencies.DATA),
-            getPrice(time, timeUnit, pricePerSecondInUsd, contractCurrencies.USD),
+            inData,
+            dataToUsd(inData, dataPerUsd),
         ]
     }, [dataPerUsd, priceCurrency, pricePerSecond, time, timeUnit])
 
-    const { account } = useWeb3Status()
+    const isValidTime = useMemo(() => !BN(time).isNaN() && BN(time).isGreaterThan(0), [time])
 
-    const isValidTime = useCallback(() => !BN(time).isNaN() && BN(time).isGreaterThan(0), [time])
-
-    const isValidPrice = useCallback(() => {
+    const isValidPrice = useMemo(() => {
         if (paymentCurrency === paymentCurrencies.ETH) {
             if (Number(priceInUsd) < MIN_UNISWAP_AMOUNT_USD) { return false }
-            return !(BN(priceInEth).isNaN() || !BN(priceInEth).isGreaterThan(0) || !BN(priceInEth).isFinite())
+            return !(BN(currentPrice).isNaN() || !BN(currentPrice).isGreaterThan(0) || !BN(currentPrice).isFinite())
         }
 
         if (paymentCurrency === paymentCurrencies.DAI) {
             if (Number(priceInUsd) < MIN_UNISWAP_AMOUNT_USD) { return false }
-            return !(BN(priceInDai).isNaN() || !BN(priceInDai).isGreaterThan(0) || !BN(priceInDai).isFinite())
+            return !(BN(currentPrice).isNaN() || !BN(currentPrice).isGreaterThan(0) || !BN(currentPrice).isFinite())
         }
 
         return true
-    }, [paymentCurrency, priceInUsd, priceInEth, priceInDai])
-
-    const getProductPrices = useCallback(async () => {
-        if (isValidTime()) {
-            setLoading(true)
-
-            const ethValue = await uniswapDATAtoETH(priceInData)
-            const daiValue = await uniswapDATAtoDAI(priceInData)
-
-            let ethUSDEquivalent = await uniswapETHtoDATA(ethValue.toString())
-            ethUSDEquivalent = ethUSDEquivalent.dividedBy(Number(dataPerUsd) || 1)
-            setEthPriceInUsd(ethUSDEquivalent.isNaN() ? 'N/A' : formatDecimals(ethUSDEquivalent, contractCurrencies.USD))
-
-            setPriceInEth(BN(ethValue).isNaN() ? 'N/A' : formatDecimals(ethValue, paymentCurrencies.ETH).toString())
-            setPriceInDai(BN(daiValue).isNaN() ? 'N/A' : formatDecimals(daiValue, paymentCurrencies.DAI).toString())
+    }, [paymentCurrency, priceInUsd, currentPrice])
+
+    const setExternalPrices = useDebounced(useCallback(async ({
```

Comment:
debouncing adds slight delay... changed the calls here to use the public node for querying

Reply:
The debounce causes the transition to be a little choppy, do you think it's strictly necessary for these sorts of calls?